packages/babel-core/src/transformation/normalize-file.js

```diff
@@ -76,7 +76,7 @@ export default function normalizeFile(
     } else if (ast.type !== "File") {
       throw new Error("AST root must be a Program or File node");
     }
-    ast = cloneDeep(ast);
+    if (options.deepCopy) ast = cloneDeep(ast);
   } else {
```

Comment:
Can you add an error here, in case the new option is used with a string input?

Reply:
Done